tools/pythonpkg/tests/fast/spark/test_spark_dataframe.py

```diff
@@ -35,8 +43,11 @@ def test_dataframe_from_list_of_tuples(self, spark):
 
         # Tuples of different sizes
         address = [(1, "14851 Jeffrey Rd", "DE"), (2, "43421 Margarita St", "NY"), (3, "13111 Siemon Ave")]
-        with pytest.raises(PySparkTypeError, match="LENGTH_SHOULD_BE_THE_SAME"):
-            df = spark.createDataFrame(address, ["id", "address", "state"])
+        if not USE_ACTUAL_SPARK:
+            # Unclear why but Spark does only raise such an error, within Java,
+            # once .collect is called.
```

Comment:
Thanks, I think I'm happy about throwing the error earlier, but we can add a `df.collect()` underneath so we both end up throwing (even if we throw earlier)

Reply:
Spark throws `Py4JJavaError`, weird... I've updated the test to better show this.